File: Sound_Assignment/codes/3.3.py
import numpy as np
import matplotlib.pyplot as plt
import scipy

n = np.linspace(0, 19, 20)
y = np.loadtxt('filter_output.dat', dtype='double')
plt.stem(n, y)
plt.title('Filter Output')
plt.xlabel('$n$')
plt.ylabel('$y(n)$')
plt.grid()
plt.savefig('../figs/3.3.png')
plt.show()

File: Sound_Assignment/codes/3.3.c

#include <stdio.h>
double x(int n) {
	if (n < 0 || n > 5) return 0;
	else if (n < 4) return n + 1;
	else return 6 - n;
}

double y(int n) {
	if (n < 0) return 0;
	else return x(n) + x(n-2) - 0.5 * y(n-1);
}

int main() {
	FILE *fp = fopen("filter_output.dat", "w");
	for (int i = 0; i < 20; i++) {
		fprintf(fp, "%lf\n", y(i));
	}
	fclose(fp);
	return 0;
}
